### PR TITLE
CAM-418 aust my cases experiment

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Continue presenter experiment by externalizing React component logic to be tested independent of React.

# Major Changes

* Added humble function for React useState
* Added testing utility to mock the humble useState
* Externalized state and actions from the StaffAssignmentRow component
* Wrote tests for the externalized state and actions.

# Testing/Validation

Increased test coverage for the StaffAssignmentRow component to 100%

### CODE REVIEW

File: user-interface/src/lib/hooks/UseState.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { Dispatch, SetStateAction } from 'react';
+
+// <S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
+// <S = undefined>(initialState: S): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
```
Comment (fmaddenflx) [inline user-interface/src/lib/hooks/UseState.ts:4]: Is this for documentation purposes or ... ?

File: user-interface/src/lib/testing/testing-utilities.ts [hunk 2]
```diff
@@ -47,11 +49,47 @@ function spyOnGlobalAlert() {
...
+      }
+    },
+  ];
+};
+
+function spyOnUseState() {
+  return vi.spyOn(UseStateModule, 'useState').mockImplementation(useStateMock);
+}
+
+export const urlRegex = /https?:\/\/.*\//;
```
Comment (jamesobrooks) [inline user-interface/src/lib/testing/testing-utilities.ts:86]: We'll need to take this out.

File: user-interface/src/lib/testing/testing-utilities.ts
Comment (fmaddenflx) [inline user-interface/src/lib/testing/testing-utilities.ts:1]: I like this.  Pretty cool.

**Review by fmaddenflx — APPROVED** [review]

File: user-interface/src/lib/hooks/UseState.ts (hunk 1 above)
Comment (btposey) [reply user-interface/src/lib/hooks/UseState.ts:4]: Not really. These are function type definitions from React.

**Pull request merged**